src/screens/MyWorkouts/WorkoutSearchBar.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { useEffect, useState, useRef } from 'react';
+import { View, Text, TouchableOpacity } from 'react-native';
+import { ScrollView, TextInput } from 'react-native-gesture-handler';
+import { supabase } from '../../utils/supabaseClient';
+
+/**
+ * @param selectedExercise.exercise //value set to exercise pressed
+ * @param selectedExercise.isOk //if true, then value of exercise is valid, otherwise disregard
+ */
+export type SearchBarWorkoutProps = {
+  selectedExercise: { exercise: string | undefined; isOk: boolean | undefined };
+};
+
+const SearchBarWorkout: React.FC<SearchBarWorkoutProps> = (prop: SearchBarWorkoutProps) => {
+  const [allExercises, setAllExercises] = useState<string[]>([]);
+  const [searchInput, setSearchInput] = useState('');
+  const recentlyUsedExercise = useRef('');
```

Comment:
I think once this component unmounts we'll lose lose this local data. It could be worth it to use `AsyncStorage` in the future to persist this on the device, but for this PR that's unnecessary.